### PR TITLE
Misc. implementations

### PR DESCRIPTION
**Pull request by dmirmilshteyn**
Redo of #813. I added some more implementations I've done since then,too.

### CODE REVIEW

File: JSIL.Libraries/Includes/Bootstrap/Core/Classes/System.Collections.Concurrent.ConcurrentQueue.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    $.Method({ Static: false, Public: true }, "TryDequeue",
+      (new JSIL.MethodSignature($.Boolean, [$jsilcore.TypeRef("JSIL.Reference", [new JSIL.GenericParameter("T", "System.Collections.Concurrent.ConcurrentQueue`1")])], [])),
+      function Dequeue(result) {
+          if (this._size > 0) {
+              var item = this._items.shift();
+              this._size -= 1;
+              result.set(item);
+              return true;
+          } else {
+              result.set(null);
```
Comment (kg) [inline JSIL.Libraries/Includes/Bootstrap/Core/Classes/System.Collections.Concurrent.ConcurrentQueue.js]: This needs to be DefaultValue not null

Comment (dmirmilshteyn) [inline JSIL.Libraries/Includes/Bootstrap/Core/Classes/System.Collections.Concurrent.ConcurrentQueue.js]: Updated

**Comment by kg** [comment]
Looks great, thanks! I'll merge once tests pass.

**Comment by kg** [comment]
Tests failed, please look at the output:

https://ci.appveyor.com/project/kg/jsil/build/1.0.7/tests
https://travis-ci.org/sq/JSIL/jobs/77973309#L9948

Looks like it might be just one problem

**Comment by dmirmilshteyn** [comment]
Forgot to include the proxy definitions for the extra DateTime methods, tests pass now

**Pull request merged**